xs/src/libslic3r/Print.cpp

```diff
@@ -1123,6 +1130,124 @@ Print::_make_brim()
+        point_round.points.push_back(Point(size_ear*-1, 0*size_ear));
+        point_round.points.push_back(Point(size_ear*-0.966, -0.26*size_ear));
+        point_round.points.push_back(Point(size_ear*-0.87, -0.5*size_ear));
+        point_round.points.push_back(Point(size_ear*-0.7, -0.7*size_ear));
+        point_round.points.push_back(Point(size_ear*-0.5, -0.87*size_ear));
+        point_round.points.push_back(Point(size_ear*-0.26, -0.966*size_ear));
+        point_round.points.push_back(Point(size_ear*0, -1*size_ear));
+        point_round.points.push_back(Point(size_ear*0.26, -0.966*size_ear));
+        point_round.points.push_back(Point(size_ear*0.5, -0.87*size_ear));
+        point_round.points.push_back(Point(size_ear*0.7, -0.7*size_ear));
+        point_round.points.push_back(Point(size_ear*0.87, -0.5*size_ear));
+        point_round.points.push_back(Point(size_ear*0.966, -0.26*size_ear));
+        
+        //create ears
+        Polygons mouse_ears;
+        for (Point pt : pt_ears) {
+            mouse_ears.push_back(point_round);
+            mouse_ears.back().translate(pt);
+        }
+        
+        //intersection
+        Polylines lines = intersection_pl(union_pt_chained(loops), mouse_ears);
+        
+        //reorder them
+        Polylines lines_sorted;
+        Polyline* previous = NULL;
+        Polyline* best = NULL;
+        double best_dist = -1;
+        size_t best_idx = 0;
+        while (lines.size() > 0) {
+            if (previous == NULL) {
+                lines_sorted.push_back(lines.back());
+                previous = &lines_sorted.back();
+                lines.erase(lines.end() - 1);
+            } else {
+                best = NULL;
+                best_dist = -1;
+                best_idx = 0;
+                for (size_t i = 0; i < lines.size(); ++i) {
+                    Polyline &viewed_line = lines[i];
+                    double dist = viewed_line.points.front().distance_to(previous->points.front());
+                    dist = std::min(dist, viewed_line.points.front().distance_to(previous->points.back()));
+                    dist = std::min(dist, viewed_line.points.back().distance_to(previous->points.front()));
+                    dist = std::min(dist, viewed_line.points.back().distance_to(previous->points.back()));
+                    if (dist < best_dist || best == NULL) {
+                        best = &viewed_line;
+                        best_dist = dist;
+                        best_idx = i;
+                    }
+                }
+                if (best != NULL) {
+                    //copy new line inside the sorted array.
+                    lines_sorted.push_back(lines[best_idx]);
+                    lines.erase(lines.begin() + best_idx);
+                    
+                    //connect if near enough
+                    if (lines_sorted.size() > 1) {
+                        size_t idx = lines_sorted.size() - 2;
+                        bool connect = false;
```

Comment:
it is possible to check `best_dist` first and skip testing distance again.

Reply:
?

Reply:
Closest distance between lines is already available in `best_dist`

Reply:
but i need to know where to do reverse(). So i need to store more information than just best_dist...

Reply:
Yes, the tests will be necessary. But testing best_dist can skip the tests altogether.
It's just minor optimization ...

Reply:
oh, i understand